vck-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/helper/RequestParser.kt

```diff
@@ -70,43 +70,35 @@ open class RequestParser(
         extractedParams
     }
 
-    private suspend fun extractRequestObject(params: AuthenticationRequestParameters): RequestParametersFrom? =
+    private suspend fun extractRequestObject(params: AuthenticationRequestParameters): RequestParametersFrom<*>? =
         params.request?.let { requestObject ->
             parseRequestObjectJws(requestObject)
         } ?: params.requestUri?.let { uri ->
             remoteResourceRetriever.invoke(uri)
                 ?.let { parseRequestParameters(it).getOrNull() }
         }
 
-    private fun parseRequestObjectJws(requestObject: String): RequestParametersFrom? {
-        return JwsSigned.deserialize(requestObject).getOrNull()?.let { jws ->
-            val params = kotlin.runCatching {
-                jsonSerializer.decodeFromString(
-                    PolymorphicSerializer(RequestParameters::class),
-                    jws.payload.decodeToString()
-                )
-            }.getOrElse {
-                return null
-                    .apply { Napier.w("parseRequestObjectJws: Deserialization failed", it) }
+    private fun parseRequestObjectJws(requestObject: String): RequestParametersFrom<*>? {
+        return JwsSigned.deserialize<AuthenticationRequestParameters>(requestObject, vckJsonSerializer).getOrNull()
+            ?.let { jws ->
+                if (requestObjectJwsVerifier.invoke(jws)) {
+                    RequestParametersFrom.JwsSigned(jws, jws.payload)
+                } else null
+                    .also { Napier.w("parseRequestObjectJws: Signature not verified for $jws") }
             }
-            if (requestObjectJwsVerifier.invoke(jws, params)) {
-                matchRequestParameterCases(jws, params)
-            } else null
-                .also { Napier.w("parseRequestObjectJws: Signature not verified for $jws") }
-        }
     }
 
-    open fun <T> matchRequestParameterCases(input: T, params: RequestParameters): RequestParametersFrom =
+    private fun <T> matchRequestParameterCases(input: T, params: RequestParameters): RequestParametersFrom<*> =
         when (params) {
             is AuthenticationRequestParameters ->
                 when (input) {
-                    is Url -> AuthenticationRequestParametersFrom.Uri(input, params)
-                    is JwsSigned<*> -> AuthenticationRequestParametersFrom.JwsSigned(
-                        input as JwsSigned<ByteArray>,
+                    is Url -> RequestParametersFrom.Uri(input, params)
+                    is JwsSigned<*> -> RequestParametersFrom.JwsSigned(
+                        input as JwsSigned<RequestParameters>,
```

Comment:
Refactor to use only one line: `RequestParametersFrom.JwsSigned(input as JwsSigned<RequestParameters>,  params)`